src/components/CommentsList/CommentsList.tsx

```diff
@@ -1,53 +1,110 @@
 "use client";
 
-import { FC, Fragment, useContext, useMemo, useState } from "react";
-import { Comment, ICommentProps } from "./Comment";
+import { FC, HTMLAttributes, useCallback, useState } from "react";
+import { useComments } from "./hooks";
+import { Comment } from "./Comment";
 import { Button } from "../Button";
 import { CommentForm } from "./CommentForm";
-import { CommentsListContext, ICommentsListContext } from "./context";
-import { ApolloQueryResult, OperationVariables } from "@apollo/client";
+import Spinner from "@/assets/icons/spinner.svg";
 
 import styles from "./CommentsList.module.css";
+import { ReplyTo } from "./styles";
 
 interface CommentsListProps {
-    comments: ICommentProps[];
     event_id: string;
-    refetch: (variables?: Partial<OperationVariables> | undefined) => Promise<ApolloQueryResult<any>>;
 }
 
-export const CommentsList: FC<CommentsListProps> = ({ comments, event_id, refetch }) => {
-    const [replyIdForm, setReplyIdForm] = useState<string | null>(null);
+const MessageContainer: FC<HTMLAttributes<HTMLDivElement>> = ({ children, className, ...rest }) => (
+    <div className={`${styles.messageContainer} ${className ?? ""}`} {...rest}>
+        {children}
+    </div>
+);
 
-    const contextValue: ICommentsListContext = { event_id, refetch, replyIdForm, setReplyIdForm };
+export const CommentsList: FC<CommentsListProps> = ({ event_id }) => {
+    const { data, loading, error, refetch, saveComment, author_id } = useComments(event_id);
+
+    const [replyToState, setReplyToState] = useState<ReplyTo | null>(null);
+    const [parentIdState, setParentIdState] = useState<string | null>(null);
+
+    const onSaveComment = useCallback(
+        async ({ content, replyTo, parentId }: { content: string; replyTo?: ReplyTo; parentId?: string }) => {
+            const res = await saveComment({
```

Comment:
```suggestion
            const saveCommentResult = await saveComment({
```
или saveCommentResult, смотря что лучше поджодит

Reply:
saveCommentResponse?